src/main/java/com/applicaster/sport1player/Sport1PlayerAdapter.java

```diff
@@ -122,13 +145,105 @@ protected void openLoginPluginIfNeeded(final boolean isInline) {
         }
     }
 
-    protected void loadItem() {
+    private void displayValidation() {
         Intent intent = new Intent(getContext(), PresentPluginActivity.class);
         intent.putExtra(PresentPluginActivity.PLUGIN_ID_EXTRA, validationPluginId);
-        intent.putExtra(PresentPluginActivity.CALLBACK_EXTRA, (PresentPluginResultI) () -> PluginManager.getInstance());
         getContext().startActivity(intent);
     }
 
+    protected void loadItem() {
+        PlayerLoader applicasterPlayerLoader = new PlayerLoader(new ApplicaterPlayerLoaderListener(isInline));
+        applicasterPlayerLoader.loadItem();
+    }
+
+    private void processLivestreamData(String json) {
+        Type mapType = new TypeToken<LinkedTreeMap<Object, Object>>(){}.getType();
+        Map<Object, Object> data = new Gson().fromJson(json, mapType);
+
+        long now = Sport1PlayerUtils.getCurrentTime();
+
+        //  Schedule next live stream check
+        if (data.containsKey(EPG_KEY)) {
+            List<LinkedTreeMap<Object, Object>> epg = (List<LinkedTreeMap<Object, Object>>) data.get(EPG_KEY);
+            if (epg != null) {
+                Map<Object, Object> lastProgram = epg.get(epg.size() - 1);
+                if (lastProgram.containsKey(END_KEY)) {
+                    String endDate = (String) lastProgram.get(END_KEY);
+                    long endTime = Sport1PlayerUtils.dateToTimestamp(endDate);
+                    long timeout = (endTime - now - 10) * 1000;
+                    if (timeout > 0) {
+                        //  set countdown to next live stream data check
+                        new CountDownTimer(timeout, timeout) {
+
+                            @Override
+                            public void onTick(long millisUntilFinished) {
+                            }
+
+                            @Override
+                            public void onFinish() {
+                                getJSON(livestreamUrl);
+                            }
+                        }.start();
+                    }
+                }
+            }
+        }
+
+        //  Check for age restriction
+        if (data.containsKey(AGE_RESTRICTION_START_KEY) && data.containsKey(AGE_RESTRICTION_END_KEY)) {
+            long ageRestrictionStart = (long)(double)data.get(AGE_RESTRICTION_START_KEY);
```

Comment:
2 castings?

Reply:
yes, for some reason timestamps like 1568216969, are parsed as double from JSON and it crashes.